### DISCUSSION TITLE
I need help regarding this API's legality.

### DISCUSSION THREAD

**Discussion by BlazingProgrammer** [timdorr/tesla-api#631] [Q&A]
This is addresed to the creator of this API (timdorr, as far as I know)

I am working on an app for the App Store using this API, however, Apple is asking me for documented proof that I can control Tesla vehicles.
I know this might sound a bit far fetched, but do you have any authorization from Tesla that I could use for my app?
If so, thank you very much in advance!

**Comment by timdorr** [comment]
I didn't create the API, I just documented it. It's a Tesla creation. There are other Tesla apps in the App Store, so presumably, there's a way to present to Apple that you have the right to access the API without Tesla officially supporting it.

**Reply by BlazingProgrammer** [reply]
Thank you very much for your help!
If you have any more suggestions regarding how I could get this permission from Tesla, you can always contact me at scheauianis@yahoo.com!

**Comment by BlazingProgrammer** [comment]
Thank you for the response, but I should explain things more clearly.

Message ID: ***@***.***>



I am trying to publish an app on the App Store that can control Tesla vehicles, but Apple keeps asking me from proof in the form of documents from Tesla that I am allowed to control their vehicles with my app (Like Tessie, Nikola for Tesla, etc...).If you have any experience as to how I could get past this or get in contact with Tesla for the permission Apple is asking me to give them, that would mean the world to me.Thank you for your responses and patience!

**Comment by itsMeDavidV** [comment]
You don't need special permission to access an open API. Using the Tesla API as it is intended with the proper authorizations is not hacking or unauthorized use.

Unfortunately, I cannot advise you further other than to lay out the fact that it is an open API that anyone with a Tesla account can access.

Apple is trying to protect against _**unauthorized**_ use of others' APIs.

**Comment by BlazingProgrammer** [comment]
Well, I did show them the MIT License, and even told them it is for the API my app uses, but they still told me I need permission from Tesla.
